tests/test_nblist.py

```diff
@@ -94,37 +95,108 @@ def build_reference_ixn_list(coords: NDArray, box: NDArray, cutoff: float) -> Li
+    np.testing.assert_array_equal(
+        np.sort(np.concatenate([col_idxs, row_idxs])),
+        identity_idxs,
+    )
 
-    col_coords = np.expand_dims(coords[:num_host_atoms], axis=0)
+    col_coords = coords[col_idxs]
+    col_coords = np.expand_dims(col_coords, axis=0)
     # Compute the reference interactions of the ligand
     ref_ixn_list = []
-    row_atoms = coords[num_host_atoms:].shape[0]
-    num_blocks = (row_atoms + block_size - 1) // block_size
-    box_diag = np.diagonal(box)
+    all_row_coords = coords[row_idxs]
+    row_length = all_row_coords.shape[0]
+    num_blocks = (row_length + block_size - 1) // block_size
+
     for rbidx in range(num_blocks):
-        row_start = num_host_atoms + (rbidx * block_size)
-        row_end = min(num_host_atoms + ((rbidx + 1) * block_size), N)
-        row_coords = coords[row_start:row_end]
+        row_start = rbidx * block_size
+        row_end = min((rbidx + 1) * block_size, N)
+        row_coords = all_row_coords[row_start:row_end]
         row_coords = np.expand_dims(row_coords, axis=1)
         deltas = image_coords(row_coords - col_coords, box_diag)
 
         dij = np.linalg.norm(deltas, axis=-1)
         # Since the row and columns are unique, don't need to handle duplicates
         idxs = np.argwhere(np.any(dij < cutoff, axis=0))
+        # Get back the column indices that are ixns
+        idxs = col_idxs[idxs.reshape(-1)]
         ref_ixn_list.append(idxs.reshape(-1).tolist())
     return ref_ixn_list
 
 
-def test_neighborlist():
+def assert_ixn_lists_are_equal(ref_ixn, test_ixn):
+    for bidx, (a, b) in enumerate(zip(ref_ixn, test_ixn)):
+        if sorted(a) != sorted(b):
+            print("TESTING bidx", bidx)
+            print(sorted(a))
+            print(sorted(b))
+        np.testing.assert_equal(sorted(a), sorted(b))
 
-    water_coords = get_water_coords(3, sort=False)
-    for size in [35, 64, 129, 1025, 1259, 2029]:
 
+def test_nblist_row_indices_are_order_independent():
+    D = 3
+    cutoff = 1.0
+    padding = 0.1
+    sizes = [35, 64, 129, 1025, 1259, 2029]
+    max_size = max(sizes)
+    water_coords = get_water_coords(D, sort=False)
+    nblists = [custom_ops.Neighborlist_f32(max_size), custom_ops.Neighborlist_f64(max_size)]
+    for size in sizes:
+        print("testing size:", size)
+
+        np.random.seed(1234)
+        water_idxs = np.random.choice(np.arange(water_coords.shape[0]), size, replace=False)
+        coords = water_coords[water_idxs]
+        diag = np.amax(coords, axis=0) - np.amin(coords, axis=0) + padding
+        box = np.diag(diag)
+
+        atom_idxs = np.random.choice(np.arange(coords.shape[0]), size // 2, replace=False)
+        atom_idxs = atom_idxs.astype(np.uint32)
+
+        reference_ixns = build_reference_ixn_list_with_subset(coords, box, cutoff, atom_idxs)
+        # Shuffle idxs, should still have the same set of interactions
+        shuffled_idxs = atom_idxs.copy()
+        np.random.shuffle(shuffled_idxs)
+
```

Comment:
nit: assert that test is not a no-op?
```python
assert shuffled_idxs != atom_idxs
```

Reply:
They are, used to debug if this test fails. Were in the old tests, just refactored into a function